compiler/qsc_frontend/src/compile/preprocess.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use core::str::FromStr;
+use qsc_ast::{
+    ast::{Attr, ExprKind, Namespace},
+    mut_visit::MutVisitor,
+};
+
+use super::Target;
+
+pub(super) struct Conditional {
+    pub(super) target: Target,
+}
+
+impl MutVisitor for Conditional {
+    fn visit_namespace(&mut self, namespace: &mut Namespace) {
+        namespace.items = namespace
+            .items
+            .iter()
+            .filter_map(|item| {
+                if matches_target(&item.attrs, self.target) {
+                    Some(item.clone())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>()
+            .into_boxed_slice();
+    }
+}
+
+fn matches_target(attrs: &[Box<Attr>], target: Target) -> bool {
+    attrs.iter().all(|attr| {
+        if attr.name.name.as_ref() == "TargetProfile" {
+            if let ExprKind::Paren(inner) = attr.arg.kind.as_ref() {
+                match inner.kind.as_ref() {
+                    ExprKind::Path(path) => match Target::from_str(path.name.name.as_ref()) {
+                        Ok(t) => t == target,
+                        Err(_) => true,
+                    },
+                    _ => true,
+                }
+            } else {
+                true
+            }
+        } else {
+            true
+        }
```

Comment:
If this nested `if` is not expected to be extended in the future, then it looks like a candidate for merging the two `if`s (if merging the `if` and `if let` is supported by Rust):
```rs
        if    (attr.name.name.as_ref() == "TargetProfile") 
           || (let ExprKind::Paren(inner) = attr.arg.kind.as_ref()) {
            . . .
        } else {
            true
        }
```

Reply:
Yes, though unfortunately that is not valid Rust code at the moment:
```
error[E0658]: `let` expressions in this position are unstable
  --> compiler\qsc_frontend\src\compile\preprocess.rs:35:51
   |
35 |         if attr.name.name.as_ref() == "Target" && let ExprKind::Paren(inner) = attr.arg.kind.as_ref() {
   |                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
```